specs/governance/mint-manager.md

```diff
@@ -0,0 +1,309 @@
+MUST enforce that the requested mint amount does not exceed `(totalSupply * MINT_CAP) / DENOMINATOR`, which equals
+2% of the current total supply. This ensures that token inflation is bounded and predictable.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is violated, the owner could mint an unlimited number of tokens, leading to:
+- Severe token dilution for existing holders
+- Loss of governance voting power for existing token holders
+- Destruction of the token's economic value
+- Complete loss of trust in the protocol's governance system
+
+### iMM-002: Time-based minting restriction
+
+After the first mint, subsequent minting operations can only occur after 365 days have elapsed since the previous mint.
+
+**Full Description:**
+Once `mintPermittedAfter` has been set to a non-zero value, any call to `mint()` MUST revert if
+`block.timestamp < mintPermittedAfter`. This ensures that minting operations are rate-limited to at most once per
+year, preventing rapid inflation even if the owner attempts multiple mints.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is violated, the owner could:
+- Mint the maximum 2% multiple times in rapid succession
+- Cause uncontrolled inflation far exceeding the intended 2% annual rate
+- Undermine the predictability and transparency of the token supply schedule
+- Violate the expectations of token holders regarding inflation rates
+
+### iMM-003: Exclusive minting authority
+
+Only the contract owner can successfully call the `mint()` function to create new governance tokens.
+
+**Full Description:**
+The `mint()` function MUST be protected by the `onlyOwner` modifier, ensuring that only the address returned by
+`owner()` can execute minting operations. Any call from a non-owner address MUST revert.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is violated:
+- Unauthorized parties could mint tokens without governance approval
+- The entire governance system would be compromised
+- Token supply would become unpredictable and uncontrolled
+- The economic security of the protocol would be destroyed
+
+### iMM-004: Ownership transfer control
+
+Only the current contract owner can transfer ownership of the GovernanceToken to a new MintManager.
+
+**Full Description:**
+The `upgrade()` function MUST be protected by the `onlyOwner` modifier, ensuring that only the current owner can
+initiate an upgrade to a new `MintManager` implementation. This prevents unauthorized parties from taking control of
+the token minting authority.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is violated:
+- Attackers could transfer ownership to a malicious contract
+- The governance system would lose control over token minting
+- A malicious MintManager could mint unlimited tokens or implement harmful policies
+- The protocol's governance would be permanently compromised
+
+### iMM-005: Valid successor requirement
```

Comment:
devinai Prefer an assumption that the successor will be valid, hard to maintain as invariant